dataland-e2etests/src/test/kotlin/org/dataland/e2etests/utils/ApiAccessor.kt

```diff
@@ -122,7 +124,32 @@ class ApiAccessor {
                 )
             }
         }
-        return listOfUploadInfo
+        if (!ensureQaPassed) return listOfUploadInfo
+        return ensureQaCompleted(listOfUploadInfo)
+    }
+
+    private fun ensureQaCompleted(listOfUploadInfo: MutableList<UploadInfo>): MutableList<UploadInfo> {
+        // Check after upload to ensure that dummy QA is passed and wait until it is
+        val maxQaPassedYetRetries = 100; val sleepIfQaNotPassedYetMs: Long = 50
+
+        val result = mutableListOf<UploadInfo>()
+        repeat(maxQaPassedYetRetries) {
```

Comment:
Isn't awaitility what you want here?
https://github.com/awaitility/awaitility/wiki/Usage#simple

Reply:
I would want to also achieve that the UploadInfo objects are updated with their current Metadata. If I just run await until a method returns true for a list of dataIds, then the UploadInfo Objects still contain Metadata that states that the QA is Pending even though it is not.

Reply:
Has been refactored to use await

Reply:
using await now.

Reply:
ok